### ISSUE TITLE
RDLogManager Progress Pop-Up Freezes When Generating Logs

### ISSUE THREAD

**Issue by glenndrives** [ElvishArtisan/rivendell#983]
We are running Rivendell V4.3.0int2-1 on Linux Mint 21 xfce.

When generating a log from RDLogManager, after clicking on the "Create New Log" button the "Progress" bar window opens and shows the progress bar. After the log is generated the "Progress" window does not close and either becomes transparent or shows the progress bar frozen close to the end.

![image](https://github.com/user-attachments/assets/a44ba245-535a-4642-9a4d-ef75abcf29ae)

We ran RDLogManager from the CLI and got the following error:

`qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 1227, resource id: 9701124, major code: 40 (TranslateCoords), minor code: 0
`

Cheers!

Glenn

**Comment by romainmp** [comment]
Not sure if it is linked, but log generation on my v4.3 / Mint 22 is taking ages compared to previous v3.6.5 on Mint 19.3.
It's been more than an hour and progress is still at 50%. At first I thought the window crashed but later on the progress appeared.

I see one thread at 100% on one CPU with top.

**Comment by bpm1992** [comment]
> Not sure if it is linked, but log generation on my v4.3 / Mint 22 is taking ages compared to previous v3.6.5 on Mint 19.3. It's been more than an hour and progress is still at 50%. At first I thought the window crashed but later on the progress appeared.
> 
> I see one thread at 100% on one CPU with top.

I added changes to use a cache for log generation, and we're in the process of testing this.  We're seeing a 95-99% reduction in time.  So far, so good.

**Comment by dklann** [comment]
Yes! Concur! I'm running with @bpm1992 branch gmc-v4, commit a733b00 and log generation takes a fraction of what it used to take (this instance has many Scheduler Codes and clock rules).
